.github/workflows/test.yml

```diff
@@ -27,17 +27,18 @@ jobs:
         run: |
           python -m pip install --upgrade pip
           pip install .
-      - name: Test requirements Installation
+      - name: Test requirements installation
         run: |
           python otherfiles/requirements-splitter.py
           pip install --upgrade --upgrade-strategy=only-if-needed -r test-requirements.txt
       - name: Test with pytest 
         run: |
           python -m pytest . --cov=opr --cov-report=term
       - name: Upload coverage to Codecov
-        uses: codecov/codecov-action@v3
+        uses: codecov/codecov-action@v4
         with:
-            fail_ci_if_error: false
+            fail_ci_if_error: true
+            token: ${{ secrets.CODECOV_TOKEN }}
```

Comment:
we didn't have these lines (40,41) previously in our other projects, is this because of updating the version of the associated action?

Reply:
Codecov offers two methods for uploading coverage:

1. General token
2. Specific token

Sometimes, the first option hits a rate limit and results in an error. Codecov recommends switching to a specific token. Both PyCM and Art have been updated to use this method.